support/util/imagefetch.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"github.com/blang/semver/v4"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetHypershiftComponentImage resolves the appropriate hypershift
+// image based on the following order of precedence (from most to least
+// preferred):
+//
+// 1. The image specified by the ControlPlaneOperatorImageAnnotation on the
+//    HostedControlPlane resource itself
+// 2. The hypershift image specified in the release payload indicated by the
+//    HostedControlPlane's release field
+// 3. The hypershift-operator's own image for release versions 4.9 and 4.10
+//
+// If no image can be found according to these rules, an error is returned.
+func GetHypershiftComponentImage(ctx context.Context, objectAnnotations map[string]string, releaseImage string, releaseProvider releaseinfo.Provider, hypershiftOperatorImage string, pullSecret []byte) (string, error) {
```

Comment:
"ComponentImage" is the CPO image, right? Why was this moved around?

Reply:
It was originally called by hosted-config-operator and the control-plane-operator so moved here. We can move it back since now it's just the hypershift operator that calls it